### ISSUE TITLE
ChannelCredentials.secure certificate option documentation

### ISSUE THREAD

**Issue by matejkramny** [grpc/grpc-dart#456]
I had an issue like some others have pointed out when connecting through a custom certificate trust store.
Using it resulted in a HandshakeException (CERTIFICATE_VERIFY_FAILED). 

Digging through my local files leads me to `security_context.dart` within a `sky_engine` package (which doesn't seem to exist anywhere).
This flutter doc explains that using a PEM formatted certificate won't work on ios devices: [setTrustedCertificates](https://api.flutter.dev/flutter/dart-io/SecurityContext/setTrustedCertificates.html)

It may be worth adding this to the documentation and assist with other similar issues.

**Comment by mraleph** [comment]
@matejkramny Yeah, I agree that documentation could be somewhat improved at least to refer to the core libraries doc. Feel free to submit PR expanding it.

**Comment by davidbnk** [comment]
So how you create a channel with a self signed certificate? The ChannelCredentials constructor doesn't accept a SecurityContext and instead create its own.

**Comment by matejkramny** [comment]
@davidbnk there's [another issue with code sample](https://github.com/grpc/grpc-dart/issues/134#issuecomment-502473319) to extend `ChannelOptions` and specify how you want to dial the connection.

Btw i've tried what you're looking for and gave up with self-signed certificates. They worked fine on Android but iOS couldn't connect. Tried with DER/Pem and pkcs12 formatted der certificate to no avail.
If you find a solution let me know!

**Comment by matejkramny** [comment]
> @matejkramny Yeah, I agree that documentation could be somewhat improved at least to refer to the core libraries doc. Feel free to submit PR expanding it.

I will do, is this over here or at the grpc/grpc.io project? Thanks

**Comment by davidbnk** [comment]
> 
> 
> @davidbnk there's [another issue with code sample](https://github.com/grpc/grpc-dart/issues/134#issuecomment-502473319) to extend `ChannelOptions` and specify how you want to dial the connection.
> 
> Btw i've tried what you're looking for and gave up with self-signed certificates. They worked fine on Android but iOS couldn't connect. Tried with DER/Pem and pkcs12 formatted der certificate to no avail.
> If you find a solution let me know!
> 
> The "solution" everyone is suggesting is to add the "onBadCertificate" callback and allow bad certificates through which works for development, but not if you want to connect through the internet using a self-signed certificate

Did you follow the [new requirements](https://derflounder.wordpress.com/2019/06/06/new-tls-security-requirements-for-ios-13-and-macos-catalina-10-15/)?

**Comment by matejkramny** [comment]
@davidbnk ah ha, thanks a lot!.

I made a certificate valid for 3650 days, which is over the limit of 875 days.

btw I supplied the certificate in the ChannelOptions certificate option and passing a password for pkcs12, which worked for Android as mentioned.

**Comment by mraleph** [comment]
> I will do, is this over here or at the grpc/grpc.io project? Thanks

Over here. Just add information in a doc-comment on `ChannelCredentials.secure`.

**Comment by ChokWah** [comment]
validateBadCertificate: I override and return true. It didn't error. No any response error or success.
